tests/test_task.py

```diff
@@ -0,0 +1,163 @@
+
+    # Setup mock namelists directory in tmp_path:
+    Path(tmp_path, internal.NAMELIST_DIR).mkdir()
+    touch(Path(tmp_path, internal.NAMELIST_DIR, internal.CABLE_NML))
+    touch(Path(tmp_path, internal.NAMELIST_DIR, internal.CABLE_SOIL_NML))
+    touch(Path(tmp_path, internal.NAMELIST_DIR, internal.CABLE_VEGETATION_NML))
+
+    setup_fluxnet_directory_tree([task], root_dir=tmp_path)
+
+    # Setup mock repository that has been checked out and built:
+    Path(tmp_path, internal.SRC_DIR, "test-branch", "offline").mkdir(parents=True)
+    touch(Path(tmp_path, internal.SRC_DIR, "test-branch", "offline", internal.CABLE_EXE))
+
+    task.fetch_files(root_dir=tmp_path)
+
+    assert Path(tmp_path, internal.SITE_TASKS_DIR,
+                task.get_task_name(), internal.CABLE_NML).exists()
+    assert Path(tmp_path, internal.SITE_TASKS_DIR,
+                task.get_task_name(), internal.CABLE_VEGETATION_NML).exists()
+    assert Path(tmp_path, internal.SITE_TASKS_DIR,
+                task.get_task_name(), internal.CABLE_SOIL_NML).exists()
+    assert Path(tmp_path, internal.SITE_TASKS_DIR,
+                task.get_task_name(), internal.CABLE_EXE).exists()
+
+    # Make mock log files and output files as if benchcab has just been run:
+    touch(Path(tmp_path, internal.SITE_OUTPUT_DIR, task.get_output_filename()))
+    assert Path(tmp_path, internal.SITE_OUTPUT_DIR, task.get_output_filename()).exists()
+
+    touch(Path(tmp_path, internal.SITE_LOG_DIR, task.get_log_filename()))
+    assert Path(tmp_path, internal.SITE_LOG_DIR, task.get_log_filename()).exists()
+
+    task.clean_task(root_dir=tmp_path)
+
+    assert not Path(tmp_path, internal.SITE_TASKS_DIR,
+                    task.get_task_name(), internal.CABLE_NML).exists()
+    assert not Path(tmp_path, internal.SITE_TASKS_DIR,
+                    task.get_task_name(), internal.CABLE_VEGETATION_NML).exists()
+    assert not Path(tmp_path, internal.SITE_TASKS_DIR,
+                    task.get_task_name(), internal.CABLE_SOIL_NML).exists()
+    assert not Path(tmp_path, internal.SITE_TASKS_DIR,
+                    task.get_task_name(), internal.CABLE_EXE).exists()
+    assert not Path(tmp_path, internal.SITE_OUTPUT_DIR, task.get_output_filename()).exists()
+    assert not Path(tmp_path, internal.SITE_LOG_DIR, task.get_log_filename()).exists()
+
+
+def test_adjust_namelist_file(tmp_path):
+    """Tests for `adjust_namelist_file()`."""
+
+    # Success case: adjust cable namelist file
+    task = Task(1, "test-branch", "forcing-file.nc", "sci0", {"some_setting": True})
+    task_dir = Path(tmp_path, internal.SITE_TASKS_DIR, task.get_task_name())
+
+    setup_fluxnet_directory_tree([task], root_dir=tmp_path)
+
+    # Create mock namelist file in task directory:
+    nml = {
+        'cable': {
+            'filename': {
+                'met': "/path/to/met/file",
+            },
+        }
+    }
+
+    f90nml.write(nml, str(task_dir / internal.CABLE_NML))
```

Comment:
General comment: I think it is best to create (and remove) test files in `setup_module` and `teardown_module` subroutines. It makes more sense if the test files are used in more than one test, but still has the benefit that your tests will clean up after themselves regardless if tests pass or not. Here is an example of what I'm talking about:

https://github.com/payu-org/payu/blob/master/test/test_setup.py#L24-L69

Regardless you should clean up that namelist file you create when the tests are finished.

Reply:
Sorry if that sounded grumpy, or that the feedback was pretty limited, but I had a quick scan and that was what jumped out at me that might be useful.

Reply:
Not at all! I suppose code reviews can come off as grumpy sounding 😄. Thanks for the comment. It made me look a bit more into how the [`tmp_path`](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html) pytest fixture worked, turns out that the files only get cleaned up after 3 temporary directories are created. I've now added an explicit setup and teardown similar to what is done in the payu unit tests, see: 23dde5922c54a75107293da0750525baa7cff065